src/crypto_utils.ts

```diff
@@ -150,35 +181,112 @@ export async function buildStringInputText(
 
     let encodedStr = encoder.encode(plaintext)
 
-    let encryptedStr = new Array<{ ciphertext: bigint, signature: Buffer }>(plaintext.length)
+    let encryptedStr = new Array<{ ciphertext: bigint, signature: Uint8Array }>(plaintext.length)
 
     for (let i = 0; i < plaintext.length; i++) {
-        const {ctInt, signature} = buildInputText(BigInt(encodedStr[i]), sender, contractAddress, functionSelector)
-        encryptedStr[i] = {ciphertext: ctInt, signature}
+        const { ctInt, signature } = buildInputText(BigInt(encodedStr[i]), sender, contractAddress, functionSelector)
+        encryptedStr[i] = { ciphertext: ctInt, signature }
     }
 
     return encryptedStr
 }
 
-export function generateAesKey() {
-    return crypto.randomBytes(block_size).toString("hex")
+export function decryptUint(ciphertext: bigint, userKey: string): bigint {
+    // Convert ciphertext to Uint8Array
+    let ctArray = new Uint8Array()
+
+    let temp = new Uint8Array()
+
+    while (ciphertext > 0) {
+        temp = new Uint8Array([Number(ciphertext & BigInt(255))])
+        ctArray = new Uint8Array([...temp, ...ctArray])
+        ciphertext >>= BigInt(8)
+    }
+
+    ctArray = new Uint8Array([...new Uint8Array(32 - ctArray.length), ...ctArray])
+
+    // Split CT into two 128-bit arrays r and cipher
+    const cipher = ctArray.subarray(0, BLOCK_SIZE)
+    const r = ctArray.subarray(BLOCK_SIZE)
+
+    const userKeyBytes = encodeKey(userKey)
+
+    // Decrypt the cipher
+    const decryptedMessage = decrypt(userKeyBytes, r, cipher)
+
+    return decodeUint(decryptedMessage)
+}
+
+export function decryptString(ciphertext: Array<bigint>, userKey: string): string {
+    let decryptedStr = new Array<number>(ciphertext.length)
+
+    for (let i = 0; i < ciphertext.length; i++) {
+        decryptedStr[i] = Number(decryptUint(ciphertext[i], userKey))
+    }
+
+    let decoder = new TextDecoder()
+
+    return decoder.decode(new Uint8Array(decryptedStr))
+}
+
+export function generateAesKey(): string {
```

Comment:
AES should be capital?

Reply:
I think the convention is to only capitalise the first letter of acronyms in camel case